skxray/tests/test_diff_roi_choice.py

```diff
@@ -0,0 +1,332 @@
+                             4, 4, 5, 5, 6, 6, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+                             4, 5, 5, 5, 5, 6, 6, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+                             5, 5, 5, 6, 6, 6, 5, 5, 5, 5, 6, 6, 6, 5, 5, 5,
+                             5, 5, 5, 5, 5, 5, 5, 5, 6, 6, 6, 6, 5, 5, 5, 5,
+                             5, 5, 5, 5, 5, 6, 6, 6])
+
+    numstep_pixels_m = np.array([34, 35, 45, 57, 62, 47])
+
+    qstep_ring_val_m = np.array([[2.5, 4.5],
+                                 [5.5, 7.5],
+                                 [8.5, 10.5],
+                                 [11.5, 13.5],
+                                 [14.5, 16.5],
+                                 [17.5, 19.5]])
+
+    pixelstep_list_m = ([0, 1, 2, 3, 4, 5, 6, 9, 10, 12, 13, 15, 16, 18,  19,
+                         21, 22, 25, 31, 32, 34, 35, 37, 38, 40, 41, 43, 44,
+                         46, 47, 50, 56, 57, 59, 60, 62, 63, 65, 66, 68, 69,
+                         71, 72, 75, 81, 82, 84, 85, 87, 88, 90, 91, 93, 94,
+                         96, 97, 100, 106, 107, 109, 110, 112, 113, 115, 116,
+                         118, 119, 121, 122, 125, 131, 132, 134, 135, 137, 138,
+                         140, 141, 143, 144, 146, 147, 150, 151, 152, 153, 154,
+                         155, 156, 159, 160, 162, 163, 165, 166, 168, 169, 171,
+                         172, 176, 177, 178, 179, 180, 183, 184, 187, 188, 190,
+                         191, 193, 194, 196, 197, 207, 208, 209, 211, 212, 214,
+                         215, 216, 218, 219, 221, 222, 225, 226, 227, 228, 229,
+                         230, 231, 232, 233, 235, 236, 237, 239, 240, 242, 243,
+                         245, 246, 247, 250, 251, 252, 253, 254, 255, 256, 259,
+                         260, 261, 263, 264, 265, 267, 268, 270, 271, 283, 284,
+                         285, 287, 288, 289, 291, 292, 295, 296, 300, 301, 302,
+                         303, 304, 305, 306, 307, 308, 309, 311, 312, 313, 315,
+                         316, 317, 319, 320, 325, 326, 327, 328, 329, 330, 331,
+                         332, 335, 336, 337, 340, 341, 343, 344, 345, 358, 359,
+                         360, 361, 364, 365, 368, 369, 375, 376, 377, 378, 379,
+                         380, 381, 382, 383, 384, 385, 387, 388, 389, 390, 392,
+                         393, 400, 401, 402, 403, 404, 405, 406, 407, 408, 411,
+                         412, 413, 416, 417, 418, 434, 435, 436, 437, 440, 441,
+                         442, 450, 451, 452, 453, 454, 455, 456, 457, 458, 459,
+                         460, 463, 464, 465, 466, 475, 476, 477, 478, 479, 480,
+                         481, 482, 483, 487, 488, 489]),
+
+    assert_almost_equal(qstep_ring_val, qstep_ring_val_m)
+    assert_array_equal(numstep_pixels, numstep_pixels_m)
+    assert_array_equal(qstep_inds, np.ravel(qstep_inds_m))
+    assert_array_equal(pixelstep_list, pixelstep_list_m)
+
+
+def test_roi_rings_diff_steps():
```

Comment:
```python
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.patches as mp
from itertools import chain

def test_demo(ax, center):

        
    pixeld_list_m = ([1, 2, 3, 4, 5, 7, 8, 9, 10, 11, 13, 14, 15,
        16, 17, 18, 19, 20, 21, 23, 24, 25, 26, 28,
        29, 30, 31, 35, 36, 38, 39, 40, 41, 43, 44,
        45, 46, 50, 51, 53, 54, 55, 56, 58, 59, 60,
        61, 65, 66, 68, 69, 70, 71, 73, 74, 75, 76,
        77, 78, 79, 80, 81, 83, 84, 85, 86, 88, 89,
        91, 92, 93, 94, 95, 97, 98, 99, 100, 101, 103,
        104, 105, 111, 112, 113, 114, 115, 116, 118,
        119, 120, 121, 122, 123, 124, 125, 126, 127,
        128, 129, 130, 132, 133, 134, 135, 136, 137,
        138, 139, 140, 141, 142, 143, 144, 146, 147,
        148, 149, 150, 151, 152, 153, 154, 155, 156,
        157, 158, 161, 162, 163, 164, 165, 166, 167,
        168, 169, 170, 171, 172, 174, 175, 176, 177,
        178, 179, 188, 189, 190, 191, 192, 193, 194,
        195, 196, 197, 198, 199, 200, 201, 202, 203,
        204, 205, 206, 207, 208, 209, 210, 211, 212,
        213, 214, 215, 216, 217, 218, 219, 220, 221,
        222, 223, 224, 225, 226, 227, 228, 229, 230,
        231, 232, 233, 234, 235, 236, 237, 238, 240,
        241, 242, 243, 244, 245, 246, 247, 248, 249,
        250, 251, 252, 254, 255, 256, 257, 258, 259,
        260, 261, 262, 263, 264, 265, 268, 269, 270,
        271, 272, 273, 274, 275, 276, 277, 278, 281,
        282, 283, 284, 294, 295, 296, 297, 298, 299,
        300, 301, 302, 303, 304, 305, 306, 307, 308,
        309, 310, 311, 312, 313, 314, 315, 316, 317,
        318, 319, 320, 321, 322, 323, 324, 325, 326,
        327, 328, 330, 331, 332, 333, 334, 335, 336,
        337, 338, 339, 340, 341, 345, 346, 347, 348,
        349, 350, 351, 352, 353, 354])
             
    qd_inds_m = np.array([1, 1, 1, 1, 1, 2, 2, 2, 3, 3, 4, 4, 1, 1,
        1, 1, 1, 1, 1, 2, 2, 3, 3, 4, 4, 1, 1, 1,
        1, 2, 2, 3, 3, 4, 4, 1, 1, 1, 1, 2, 2, 3,
        3, 4, 4, 1, 1, 1, 1, 2, 2, 3, 3, 4, 4, 1,
        1, 1, 1, 1, 1, 1, 2, 2, 3, 3, 4, 4, 1, 1,
        1, 1, 1, 2, 2, 2, 3, 3, 4, 4, 2, 2, 2, 2,
        3, 3, 3, 4, 4, 2, 2, 2, 2, 2, 2, 2, 2, 3,
        3, 3, 4, 4, 5, 2, 2, 2, 2, 2, 2, 2, 3, 3,
        3, 4, 4, 4, 5, 3, 3, 3, 3, 3, 3, 3, 3, 3,
        4, 4, 5, 5, 3, 3, 3, 3, 3, 3, 3, 3, 4, 4,
        4, 5, 5, 5, 4, 4, 4, 5, 5, 5, 6, 4, 4, 4,
        4, 4, 4, 4, 4, 4, 4, 5, 5, 5, 6, 6, 4, 4,
        4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 6, 6, 6, 5,
        5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 6, 6, 6, 5,
        5, 5, 5, 5, 5, 5, 5, 5, 6, 6, 6, 6, 7, 6,
        6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 7, 7, 6, 6,
        6, 6, 6, 6, 6, 6, 6, 7, 7, 7, 7, 7, 7, 7,
        7, 7, 8, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7,
        7, 8, 8, 8, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7,
        8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8,
        8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8])
            

    tt = np.ones((20, 25)).ravel() * np.nan
    tt[pixel_list_m] = q_inds_m
    
    ax.cla()
    ax.set_aspect('equal')
    im = ax.imshow(tt.reshape(20, 25), cmap='Paired', interpolation='nearest')

    
    qd_ring_val_m = np.array([[2., 4.],
                              [4.4, 6.4],
                              [6.6, 8.6],
                              [9.1, 11.1],
                              [11.5, 13.5],
                              [13.5, 15.5],
                              [16.1, 18.1],
                              [18.3, 20.3]])

    [ax.add_artist(mp.Circle(center, radius=j, lw=3, facecolor='none', edgecolor='k')) for j in qd_ring_val_m[:, 0]]
    [ax.add_artist(mp.Circle(center, radius=j, lw=3, facecolor='none', edgecolor='k', linestyle='dashed')) for j in qd_ring_val_m[:, 1]]
    Y, X = np.meshgrid(range(20), range(25))
    ax.plot(X.ravel(), Y.ravel(), 'x')
    ax.set_title(str(center))


fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 6), tight_layout=True)
test_demo(ax1, (3, 3))
test_demo(ax2, (4, 4))
```

The dashed lines are the outside edges.
![so](https://cloud.githubusercontent.com/assets/199813/6138095/3d3d194c-b14c-11e4-9e2d-85c26982f2d6.png)